### ISSUE TITLE
La7d "Connection error"

### ISSUE THREAD

**Issue by OldTimeM** [Tundrak/IPTV-Italia#315]
Ciao.
La7d non funziona su lista https://github.com/Tundrak/IPTV-Italia/raw/main/iptvitaplus.m3u

**Informazioni sul dispositivo**
 - Tipo di dispositivo: [Computer, Smart TV]
 - Programma utilizzato [M3U IPTV]

**Comment by Trifone46** [comment]
Ciao La7d attualmente si puo vedere solo tramite il link ufficiale https://www.la7.it/live-la7d non ci sono link disponibili esterni il link presente nella lista non e piu disponibile da tempo

**Comment by FrancescoRosi** [comment]
Il link nella lista ormai è morto da tempo. Usa questo: https://viamotionhsi.netplus.ch/live/eds/la7d/browser-dash/la7d.mpd

**Comment by Trifone46** [comment]
Ciao FrancescoRosi ho provato il link non va tanto bene si blocca spesse volte e non e fluido inbece se il finale e in m3u8 non ho problemi ho testato con altri canali c'è modo di cambiare l estensione finale sul finale di la7d ? Questo e il sito ufficiale Riguardo la diretta La7d https://www.la7.it/live-la7d un utente a detto Andrebbe estratto il flusso HbbTV, io ho provato ma non riesco a risalire al sito hbbtv la7d per estrarre il flusso

**Comment by FrancescoRosi** [comment]
> Ciao FrancescoRosi ho provato il link a me non va tanto bene si blocca spesse volte e non e fluido inbece se il finale e in m3u8 non ho problemi ho testato con altri canali c'è modo di cambiare l estensione finale sul finale di la7d ? Questo e il sito ufficiale Riguardo la diretta La7d https://www.la7.it/live-la7d un utente a detto Andrebbe estratto il flusso HbbTV, io ho provato ma non riesco a risalire al sito hbbtv la7d per estrarre il flusso

L'ho trovato in HLS (.m3u8), fammi sapere se funziona meglio: https://viamotionhsi.netplus.ch/live/eds/la7d/browser-HLS8/la7d.m3u8

**Comment by Trifone46** [comment]
Ciao FrancescoRosi tramite quel formato HLS (.m3u8) non ho avuto problemi ti confermo che il link funziona bene il canale si vede bene e fluido non si blocca per nulla grazie mille per essere riuscito a trovare un link in formato HLS (.m3u8) che tu sappia questo link a una data di scadenza ?

**Comment by FrancescoRosi** [comment]
> Ciao FrancescoRosi tramite quel formato HLS (.m3u8) non ho avuto problemi ti confermo che il link funziona bene il canale si vede bene e fluido non si blocca per nulla grazie mille per essere riuscito a trovare un link in formato HLS (.m3u8) che tu sappia questo link a una data di scadenza ?

No, non ha una data di scadenza, quindi smetterà di funzionare solo se l'azienda cambia il proprio sistema di link :)

**Comment by Jianluca** [comment]
Quel link è di un servizio a pagamento estero, non è meglio estrarlo da HbbTV?

**Comment by FrancescoRosi** [comment]
> Quel link è di un servizio a pagamento estero, non è meglio estrarlo da HbbTV?

Il link dell'app HbbTV ha qualche sistema complesso di signature e quant'altro.
`https://d1aud3ze4v205y.cloudfront.net/HBBTV/LA7D/DASH/Live.mpd`
`?Expires={timestamp Unix in secondi}` (sembrano avere una durata di circa 6 ore)
`&Signature=XXXXXXXXXXXXXXXXXXXXXXXXXXXXX~XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX~XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX-X-XXXXXXXXXXXXXXXXXXXXXXXXXXX-XXX~XXXX~XXXXXXXXXXXXXXXXXXXXXXXXXXX-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX~XXXXXXX~XXXXXXXXXX__`
`&Key-Pair-Id=XXXXXXXXXXXXX`

Io non me ne intendo di queste cose quindi non saprei minimamente cosa fare. Se c'è qualcuno di più esperto che vuole provare a reverse-engineerare il loro sistema di signature, ecco l'URL all'API che genera il link: https://www.la7.it/appPlayer/liveUrlWithFailPerApp.php?channel=la7d_live&v=1 (sì, purtroppo è tutto fatto lato server)

**Comment by Jianluca** [comment]
> > Quel link è di un servizio a pagamento estero, non è meglio estrarlo da HbbTV?
> 
> Il link dell'app HbbTV ha qualche sistema complesso di signature e quant'altro. `https://d1aud3ze4v205y.cloudfront.net/HBBTV/LA7D/DASH/Live.mpd` `?Expires={timestamp Unix in secondi}` (sembrano avere una durata di circa 6 ore) `&Signature=XXXXXXXXXXXXXXXXXXXXXXXXXXXXX~XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX~XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX-X-XXXXXXXXXXXXXXXXXXXXXXXXXXX-XXX~XXXX~XXXXXXXXXXXXXXXXXXXXXXXXXXX-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX~XXXXXXX~XXXXXXXXXX__` `&Key-Pair-Id=XXXXXXXXXXXXX`
> 
> Io non me ne intendo di queste cose quindi non saprei minimamente cosa fare. Se c'è qualcuno di più esperto che vuole provare a reverse-engineerare il loro sistema di signature, ecco l'URL all'API che genera il link: https://www.la7.it/appPlayer/liveUrlWithFailPerApp.php?channel=la7d_live&v=1 (sì, purtroppo è tutto fatto lato server)

Sono riuscito, bastava cambiare il dominio 
https://d15umi5iaezxgx.cloudfront.net/HBBTV/LA7/DASH/Live.mpd
https://d15umi5iaezxgx.cloudfront.net/HBBTV/LA7D/DASH/Live.mpd

**Comment by Trifone46** [comment]
Qualcuno riesce a trovarli con estensione m3u8

Il Mar 21 Apr 2026 21:19 Jianluca ***@***.***> ha scritto:

**Comment by Jianluca** [comment]
HbbTV non usa HLS quindi no
